sklearn/tests/test_calibration.py

```diff
@@ -401,6 +443,74 @@ def test_sigmoid_calibration():
         _SigmoidCalibration().fit(np.vstack((exF, exF)), exY)
 
 
+@pytest.mark.parametrize(
+    "clf",
+    [
+        DummyClassifier(),
+        RandomForestClassifier(n_estimators=30, random_state=42),
+        LogisticRegression(),
+    ],
+)
+@pytest.mark.parametrize(
+    "n_classes",
+    [2, 3, 4, 5],
+)
+def test_temperature_scaling(clf, n_classes):
+    """Check temperature scaling calibration"""
+    X, y = make_classification(
+        n_classes=n_classes,
+        n_clusters_per_class=1,
+        n_informative=n_classes,
+        random_state=42,
+    )
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+    clf.fit(X_train, y_train)
+    cal_clf = CalibratedClassifierCV(clf, method="temperature").fit(X_train, y_train)
+
+    y_pred = clf.predict(X_test)
+
+    y_scores = clf.predict_proba(X_test)
+
+    y_scores_cal = cal_clf.calibrated_classifiers_[0].calibrators[0].predict(y_scores)
+
+    y_pred_cal = np.argmax(y_scores_cal, axis=1)
+
+    if n_classes == 2:
+        y_scores = y_scores[:, 1]
+        y_scores_cal = y_scores_cal[:, 1]
+
+    # Accuracy is invariant under temperature scaling.
+    assert accuracy_score(y_test, y_pred) == accuracy_score(y_test, y_pred_cal)
+
+    # The optimized temperature should always be positive
+    assert 0.1 < cal_clf.calibrated_classifiers_[0].calibrators[0].beta < 10
+
+    # Refinement error should be invariant under temperature scaling.
+    # Use ROC AUC as a proxy for refinement error.
+    roc_auc_diff = np.abs(
+        roc_auc_score(y_test, y_scores_cal, multi_class="ovr")
+        - roc_auc_score(y_test, y_scores, multi_class="ovr")
+    )
+
+    assert roc_auc_diff <= 0.02
```

Comment:
Should we also add this:
`# Check that brier score has improved after calibration`